swarm/src/handler.rs

```diff
@@ -93,10 +93,10 @@ pub use select::{ConnectionHandlerSelect, IntoConnectionHandlerSelect};
 /// When a connection is closed gracefully, the substreams used by the handler may still
 /// continue reading data until the remote closes its side of the connection.
 pub trait ConnectionHandler: Send + 'static {
-    /// Custom event that can be received from the outside.
-    type InEvent: fmt::Debug + Send + 'static;
-    /// Custom event that can be produced by the handler and that will be returned to the outside.
-    type OutEvent: fmt::Debug + Send + 'static;
+    /// `FromBehaviour` event that can be received from the outside.
+    type FromBehaviour: fmt::Debug + Send + 'static;
+    /// `ToBehaviour` event that can be produced by the handler and that will be returned to the outside.
```

Comment:
```suggestion
    /// A type representing message(s) a [`ConnectionHandler`] can send to a [`NetworkBehaviour`] via [`ConnectionHandlerEvent::Custom`].
```

Let's go with this. Plus, while we are at it, can you rename `ConnectionHandlerEvent::Custom` to `ConnectionHandlerEvent::NotifyBehaviour`? That will be more consistent and hopefully easier to understand.

Reply:
@thomaseizinger Done

Reply:
Awesome, thanks!

Mind adding a changelog entry for it too?

Reply:
> Plus, while we are at it, can you rename `ConnectionHandlerEvent::Custom` to `ConnectionHandlerEvent::NotifyBehaviour`? That will be more consistent and hopefully easier to understand.

@tcoratger I noticed that we never ended up doing this. Are you up for sending a PR to do this rename as well?

Reply:
@thomaseizinger You are right, I forgot it, the PR is created: #3955.